app/controllers/insights_cloud/hits_controller.rb

```diff
@@ -23,9 +23,9 @@ def show
 
     def resolutions
       if remediation_all_selected_param
-        hits = InsightsHit.with_playbook.search_for(params[:query])
+        hits = resource_base.with_playbook.search_for(params[:query])
       else
-        hits = resource_base_search_and_page.where(id: remediation_ids_param)
+        hits = resource_base_search_and_page.with_playbook.where(id: remediation_ids_param)
```

Comment:
How does this work? `resource_base_search_and_page` expects the search query in `params[:search]`, yet the truthy branch uses `params[:query]`. Which is right?

Reply:
It depends on a switch on user side: when the user switches the "select all from all pages", we pass the `:query` param. Otherwise we use the regular paginated table parameters (`:search`, `:page` e.t.c.)

Reply:
A bit unexpected but fine